engine/src/main/java/io/camunda/zeebe/engine/processing/common/CommandDistributionBehavior.java

```diff
@@ -80,7 +80,11 @@ private <T extends UnifiedRecordValue> void distributeToPartition(
     sideEffectWriter.appendSideEffect(
         () -> {
           interPartitionCommandSender.sendCommand(
-              partition, command.getValueType(), command.getIntent(), key, command.getValue());
+              partition,
+              command.getValueType(),
+              command.getIntent(),
+              key,
+              distributionRecord.getCommandValue());
```

Comment:
🔧 For performance, we could consider making one copy of the distribution record's command value instead of copying it for each partition (line 62).